google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/CreateClusterRequest.java

```diff
@@ -0,0 +1,129 @@
+ * requests are actually handled by one of the clusters in the instance.
+ *
+ * <p>Each cluster is located in a single zone. An instance's clusters must be in unique zones that
+ * are within the same region. For example, if the first cluster is in us-east1-b, then us-east1-c
+ * is a valid zone for the second cluster. For a list of zones and regions where Cloud Bigtable is
+ * available, see <a href="https://cloud.google.com/bigtable/docs/locations">Cloud Bigtable Locations</a>.
+ *
+ *
+ * Examples:
+ *
+ * <pre>{@code
+ * // Small production instance:
+ * CreateClusterRequest clusterRequest = CreateClusterRequest.of("my-existing-instance", "my-cluster")
+ *   .setZone("us-east1-c")
+ *   .setServeNodes(3)
+ *   .setStorageType(StorageType.SSD);
+ * }</pre>
+ *
+ * @see <a href="https://cloud.google.com/bigtable/docs/instances-clusters-nodes#clusters">For more details</a>
+ */
+public final class CreateClusterRequest {
+  private final com.google.bigtable.admin.v2.CreateClusterRequest.Builder proto = com.google.bigtable.admin.v2.CreateClusterRequest
+      .newBuilder();
+  // Partial ids will be set when the project is passed to toProto
+  private final String instanceId;
+  private String zone;
+
+
+  /**
+   * Builds a new request to create a new cluster to the specified instance with the specified
+   * cluster id. */
+  public static CreateClusterRequest of(String instanceId, String clusterId) {
+    return new CreateClusterRequest(instanceId, clusterId);
+  }
+
+  private CreateClusterRequest(String instanceId, String clusterId) {
+    this.instanceId = instanceId;
+    proto.setClusterId(clusterId);
+    proto.getClusterBuilder().setDefaultStorageType(StorageType.SSD);
+  }
+
+  /**
+   * Sets the zone where the new cluster will be located. Must be different from the existing
+   * cluster.
+   */
+  @SuppressWarnings("WeakerAccess")
+  public CreateClusterRequest setZone(String zone) {
+    this.zone = zone;
+    return this;
+  }
+
+  /** Sets the type of storage used by this cluster to serve its parent instance's tables. */
+  @SuppressWarnings("WeakerAccess")
+  public CreateClusterRequest setServeNodes(int numNodes) {
+    proto.getClusterBuilder().setServeNodes(numNodes);
+    return this;
+  }
+
+  /**
+   * Sets the type of storage used by this cluster to serve its parent instance's tables.
+   * Defaults to {@code SSD}.
+   */
+  // TODO(igorbernstein2): try to avoid leaking protobuf generated enums
+  @SuppressWarnings("WeakerAccess")
+  public CreateClusterRequest setStorageType(StorageType storageType) {
+    proto.getClusterBuilder().setDefaultStorageType(storageType);
+    return this;
+  }
+
+  /**
+   * Creates the request protobuf. This method is considered an internal implementation detail and
+   * not meant to be used by applications.
+   */
+  @InternalApi
+  public com.google.bigtable.admin.v2.CreateClusterRequest toProto(ProjectName projectName) {
```

Comment:
Setting the fields is an internal implementation detail and since the toProto methods always overwrite the same fields, the side effects can only be observed if multiple threads try to reuse the same wrapper, which I don't see as a valid usecase.

Reply:
Sure, I am not even talking about consistency problem.
I feel like to***() transform X to Y without updating X.
Anyway, it is a nit.